velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -462,6 +467,42 @@ uint32_t HiveDataSink::appendWriter(const HiveWriterId& id) {
   return writerIndexMap_[id];
 }
 
+std::unique_ptr<facebook::velox::dwio::common::Writer>
+HiveDataSink::maybeCreateBucketSortWriter(
+    std::unique_ptr<facebook::velox::dwio::common::Writer> writer) {
+  if (insertTableHandle_->bucketProperty() == nullptr) {
+    return writer;
+  }
+  const auto& sortedProperty = insertTableHandle_->bucketProperty()->sortedBy();
+  if (sortedProperty.empty()) {
+    return writer;
+  }
+  std::vector<column_index_t> sortColumnIndices;
```

Comment:
How about we initialize these sorting properties once in ctor? e.g. sortColumnIndices_ and sortCompareFlags_